### PR TITLE
Add task to test custom builds

### PR DESCRIPTION
**Pull request by tsauerwein**
Replacement for https://github.com/openlayers/ol3/pull/5738

This PR adds a task that tests custom build configurations provided in the directory `test_custom_builds`.

To make the test pass, I had to add a `goog.require` for a type that is only used in a JSDoc type definition. These were removed in https://github.com/openlayers/ol3/pull/5698, but otherwise the compiler will fail.

### CODE REVIEW

**Comment by marcjansen** [comment]
I like it a lot, @tsauerwein. I know this again makes the CI target a little bit slower, but I guess it is worth the effort.

Please merge, and thanks again.

**Comment by tsauerwein** [comment]
Thanks, Marc. I will wait with the merge because I would like to get a feedback from @tschaub about the `goog.requires/provides` that I added again.

**Comment by marcjansen** [comment]
OK

**Comment by probins** [comment]
this is doing something similar to #5341 - combine?

File: test_custom_builds/workshop.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+       "externs/esrijson.js",
+       "externs/geojson.js",
+       "externs/oli.js",
+       "externs/olx.js",
+       "externs/proj4js.js",
+       "externs/tilejson.js",
+       "externs/topojson.js"
+     ],
+     "define": [
+       "goog.array.ASSUME_NATIVE_FUNCTIONS=true",
```
Comment (probins) [inline test_custom_builds/workshop.json]: this is no longer needed

File: test_custom_builds/workshop.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+       "ol.ENABLE_PROJ4JS=false",
+       "ol.ENABLE_IMAGE=false",
+       "goog.DEBUG=false"
+     ],
+     "jscomp_error": [
+       "*"
+     ],
+     "jscomp_off": [
+       "useOfGoogBase",
+       "unnecessaryCasts",
```
Comment (probins) [inline test_custom_builds/workshop.json]: ditto

File: test_custom_builds/workshop.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+       "ol.ENABLE_WEBGL=false",
+       "ol.ENABLE_PROJ4JS=false",
+       "ol.ENABLE_IMAGE=false",
+       "goog.DEBUG=false"
+     ],
+     "jscomp_error": [
+       "*"
+     ],
+     "jscomp_off": [
+       "useOfGoogBase",
```
Comment (probins) [inline test_custom_builds/workshop.json]: ditto

**Comment by tsauerwein** [comment]
> this is doing something similar to #5341 - combine?

The motivation, to check that there are no regressions with custom builds, is the same. Though I would like to do it only with a small set of example configurations, so that the check can be run in CI.

**Comment by marcjansen** [comment]
Thanks for comments @probins and thanks for taking them @tsauerwein :clap:

**Comment by tschaub** [comment]
Hmm.  It feels like we may just want to always throw the whole library at the Compiler and never use the "manage_closure_dependencies" option.

**Comment by tschaub** [comment]
Let's revisit this after transforming sources to ES modules for the 5.0 release.

**Pull request closed without merging**